src/components/editor/SinopiaResourceTemplates.jsx

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 Stanford University see Apache2.txt for license
+
+import React, {Component} from 'react'
+import BootstrapTable from 'react-bootstrap-table-next'
+const SinopiaServer = require('sinopia_server')
+const instance = new SinopiaServer.LDPApi()
+instance.apiClient.basePath = 'http://localhost:8080'
+
+class SinopiaResourceTemplates extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      isLoading: false,
+      groupData: [],
+      templatesForGroup: []
+    }
+  }
+
+  async componentDidMount() {
+    const groupPromise = new Promise((resolve) => {
+      resolve(instance.getBaseWithHttpInfo())
+    }).then((data) => data).catch(() => {})
+
+    await this.fulfillGroupPromise(groupPromise).then(async () => {
+      const groupDataPromise = new Promise((resolve) => {
+        this.state.groupData.map((group) => {
+          const name = this.resourceToName(group)
+          resolve(instance.getGroupWithHttpInfo(name))
+        })
+      }).then((data) => data).catch(() => {})
+      await this.fulfillGroupDataPromise(groupDataPromise)
+    })
+  }
+
+
+  fulfillGroupPromise = (promise) => {
+    promise.then((data) => {
+      const contains = [].concat(data.response.body.contains)
+      this.setState({groupData: contains})
+    }).catch((error) => {
+      this.setState({message: error})
```

Comment:
this is the line that may not be tested ... and does display something to end user.

Reply:
test added